testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -74,4 +117,82 @@ def test_slicing(self):
         with tempdir.in_tempdir():
             CA1 = self._run_ContactAnalysis1(start=start, stop=stop, step=step)
             frames = np.arange(self.universe.trajectory.n_frames)[start:stop:step]
-            self.assertEqual(CA1.timeseries.shape[1], len(frames))
+            self.assertEqual(len(CA1.timeseries), len(frames))
+
+
+    def test_math_folded(self):
+
+        u = self.folded
+
+        # read the text files
+        data = [l.split() for l in open(contacts_file).readlines()]
+        # convert to 0-based indexing
+        data = [ (int(i)-1, int(j)-1, float(d))  for i, j, d in data]
+        # get r and r0
+        data = [ (np.linalg.norm(u.atoms[i].pos - u.atoms[j].pos), d)  for i, j, d in data]
+        data = np.array(data)
+
+        r = data[:,0]
+        r0 = data[:,1]
+
+        beta = 5.0
+        lambda_constant = 1.8
+
+        Q = 1/(1 + np.exp(beta*(r - lambda_constant * r0)))
+
+        assert_almost_equal(Q.mean(),  1.0, decimal=3)
+
+    def test_math_unfolded(self):
+
+        u = self.unfolded
+
+        # read the text files
+        data = [l.split() for l in open(contacts_file).readlines()]
+        # convert to 0-based indexing
+        data = [ (int(i)-1, int(j)-1, float(d))  for i, j, d in data]
+        # get r and r0
+        data = [ (np.linalg.norm(u.atoms[i].pos - u.atoms[j].pos), d)  for i, j, d in data]
+        data = np.array(data)
+
+        r = data[:,0]
+        r0 = data[:,1]
+
+        beta = 5.0
+        lambda_constant = 1.8
+
+        Q = 1/(1 + np.exp(beta*(r - lambda_constant * r0)))
+
+        assert_almost_equal(Q.mean(),  0.0, decimal=1)
+
+    def test_villin_folded(self):
```

Comment:
I just looked into the comments from Quantified Code.Very nice tool to spot stuff like this. You can see the details of the report at the bottom of the PR.